dask/array/core.py

```diff
@@ -1086,10 +1087,10 @@ def store(
         )
 
     if isinstance(regions, tuple) or regions is None:
-        regions = [regions]
+        regions = [regions]  # type: ignore
```

Comment:
So regions can be a Collection[None]. This should be reflected in the type declaration; with the type: ignore you're instead completely neutering mypy checks for the use case below.
```suggestion
        regions = [regions]
```

Reply:
Thanks for catching this -- I'm getting hung on this section. Removing the ignore comment and adding `Collection[None]` to the type declariation brings up:

```
```

Reply:
It works fine for me with
```python
    regions: tuple[slice, ...] | Collection[tuple[slice, ...] | None] | None = None,
```
It won't work with
```python
    regions: tuple[slice, ...] | Collection[tuple[slice, ...]] | Collection[None] | None = None,
```